evm_loader/program/src/evm/buffer.rs

```diff
@@ -20,41 +20,57 @@ enum Inner {
         data: *mut u8,
         range: Range<usize>,
     },
+    ByteVec(Vec<u8>),
     AccountUninit {
         key: Pubkey,
         range: Range<usize>,
     },
 }
 
-#[derive(Debug)]
+/// SAFETY: It's safe to send `Inner` to another thread because its pointer to account is private
+unsafe impl Send for Inner {}
+/// SAFETY: We can mark `Inner` as `Sync` because it's read-only
+unsafe impl Sync for Inner {}
```

Comment:
This class are not safe to be Send or Sync

Reply:
Rolled back.